src/main/java/tsml/classifiers/interval_based/TSF.java

```diff
@@ -614,11 +796,56 @@ private void copyParameters(TSF other){
     public long getTrainContractTimeNanos(){
             return trainContractTimeNanos;
     }
-/**
- * @param ins to classifier
- * @return array of doubles: probability of each class 
- * @throws Exception 
- */   
+
+    /**
+     * @param ins TimeSeriesInstance to classifier
+     * @return array of doubles: probability of each class
+     */
+    @Override
+    public double[] distributionForInstance(TimeSeriesInstance ins) throws Exception {
+        double[] d = new double[ins.getClassLabels().length];
+        List<Double> tempSeries = ins.get(0).getSeries();
+        double[] series = tempSeries.stream().mapToDouble(data -> data).toArray();
+
+        for (int i = 0; i < trees.size(); i++) {
```

Comment:
There are also functions to extract nice and easy intervals and slices of the data to save you the hassle of doing this kind of indexing.